### PR TITLE
openfunds v2

### PR DESCRIPTION
**Pull request by 0x0ece**

### CODE REVIEW

File: anchor/programs/glam/src/state/fund.rs [hunk 1]
```diff
@@ -6,6 +6,149 @@ pub const MAX_FUND_NAME: usize = 50;
...
+    Percentage(u32), // 100% = 1_000_000
+    URI(String),
+    Timestamp(i64),
+    VecPubkey(Vec<Pubkey>),
+    VecU32(Vec<u32>),
+}
+type Map = Vec<(String, Value)>;
+
+impl From<OpenfundFund> for Map {
+    fn from(of: OpenfundFund) -> Self {
```
Comment (yurushao) [inline anchor/programs/glam/src/state/fund.rs:31]: do we need to implement common map functions like `get` and `put`?

https://crates.io/crates/map_vec provides Map and Set APIs backed by Vec. we can probably use it or customize it.

Comment (yurushao) [reply anchor/programs/glam/src/state/fund.rs:31]: wait if we use an "advanced" data structure anchor currently doesn't support we'll have to solve 2 problems (1) serializing/deserializing (2) idl translation

File: anchor/tests/glam_crud.spec.ts [hunk 2]
```diff
@@ -31,12 +31,18 @@ describe("glam_crud", () => {
     program.programId
   );
   const fundUri = `https://devnet.glam.systems/#/products/${fundPDA.toBase58()}`;
+  const openfundUri = `https://api.glam.systems/openfund/${fundPDA.toBase58()}`;
```
Comment (yurushao) [inline anchor/tests/glam_crud.spec.ts:35]: We need different uris?

File: anchor/src/index.ts [hunk 3]
```diff
@@ -1 +1 @@
-export * from './glam-exports';
+export * from './glamExports';
```
Comment (yurushao) [inline anchor/src/index.ts:1]: Where is `glamExports` defined?

File: anchor/programs/glam/src/state/fund.rs [hunk 4]
```diff
@@ -6,6 +6,178 @@ pub const MAX_FUND_NAME: usize = 50;
...
+
+pub struct OpenfundCompany {
+    // #[openfund("OFST001000", tags=["Essential"])]
+    pub FundGroupName: String,
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct OpenfundFund {
+    // #[openfund("OFST010010", tags=["Essential"])]
+    pub FundDomicileAlpha2: String,
```
Comment (yurushao) [inline anchor/programs/glam/src/state/fund.rs:154]: snake case for all fields?

File: anchor/programs/glam/src/instructions/manager_v2.rs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+use spl_token_2022::{extension::ExtensionType, state::Mint as StateMint};
+
+use crate::error::ManagerError;
+use crate::state::fund::*;
+
+#[derive(Accounts)]
+#[instruction(fund_name: String)]
+pub struct InitializeFundV2<'info> {
+    #[account(init, seeds = [b"fund".as_ref(), manager.key().as_ref(), fund_name.as_ref()], bump, payer = manager, space = FundV2::INIT_SIZE)]
+    pub fund: Box<Account<'info, FundV2>>,
```
[the remark this answers is not in the document]
Comment (0x0ece) [reply anchor/programs/glam/src/instructions/manager_v2.rs:13]: I changed it to an arbitrary created key. We can set it up automagically in the js sdk.

File: anchor/src/index.ts (hunk 3 above)
Comment (0x0ece) [reply anchor/src/index.ts:1]: Just renamed :)

File: anchor/programs/glam/src/state/fund.rs (hunk 4 above)
Comment (0x0ece) [reply anchor/programs/glam/src/state/fund.rs:154]: ++ these are now enums, so should be good

File: anchor/tests/glam_crud.spec.ts (hunk 2 above)
Comment (0x0ece) [reply anchor/tests/glam_crud.spec.ts:50]: I think it makes it a better demo, but need a bit more time to arrive to the ah-ah moment :)

File: anchor/src/client.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+import { GlamClientConfig } from "./clientConfig";
+
+type FundModel = IdlTypes<Glam>["FundModel"];
+
+export class GlamClient {
+    provider: anchor.Provider;
+    program: GlamProgram;
+    programId: PublicKey;
+
+	public constructor(config?: GlamClientConfig) {
```
Comment (yurushao) [inline anchor/src/client.ts:22]: bad indentation

File: api/app.yaml [hunk 7]
```diff
@@ -1 +0,0 @@
-runtime: nodejs20
```
Comment (yurushao) [inline api/app.yaml:1]: why deleting this file? we need it in order to deploy to app engine.

Comment (0x0ece) [reply api/app.yaml:1]: I didn't delete it, it was moved to api_v0. Will move it back soon :)

**Pull request closed without merging**